tests/polkadot_tests/calls.rs

```diff
@@ -1203,3 +1221,102 @@ fn try_catch_transfer_fail() {
     assert_eq!(runtime.output(), Vec::<u8>::new().encode());
     assert!(runtime.debug_buffer().contains("seal_instantiate=5"));
 }
+
+#[test]
+fn try_catch_panic() {
+    let mut runtime = build_solidity(
+        r#"contract A {
+        function a() public payable returns (uint) {
+            B b = new B();
+            try b.b(0) {} catch Panic(uint code) {
+                return code;
+            }
+            revert("didn't catch");
+        }
+    }
+
+    contract B {
+        function b(uint div) public pure returns(uint) {
+            return 123 / div;
+        }
+    }
+    "#,
+    );
+
+    runtime.function("a", vec![]);
+    let expected_output = U256::from(PanicCode::DivisionByZero as u8).encode();
+    assert_eq!(runtime.output(), expected_output);
+}
+
+#[test]
+fn try_catch_different_errors() {
+    let mut runtime = build_solidity(
+        r#"contract A {
+        function a(uint div) public payable returns (uint) {
+            B b = new B();
+            try b.b(div) returns (uint) {
+                return 3;
+            } catch Error(string) {
+                return 1;
+            } catch Panic(uint) {
+                return 0;
```

Comment:
I think it is good to ensure that the string and the integer are properly decoded.

Reply:
It's tested in other places, but I added the assert anyways